### PR TITLE
Addition of the computation of aerodynamic derivatives

### LINKED ISSUE #181: Add unit for CnBeta fuselage

### LINKED ISSUE #186: Remove math import
OpenMDAO variables are numpy arrays so it would be better to favor the use of numpy instead of math for mathematic operations

### PR DESCRIPTION
**Pull request by florentLutz**
This PR is based on the work of @hugogarcia06 on the computation of the lateral aerodynamic derivatives.
Here is a few remarks on their use:
- Some of these coefficients depend on lifting surface dihedral and twist angle. They have been added as inputs but they are only used in those components for now (no effects on the longitudinal derivatives).
- Some of these components depends on the flight conditions. It reuse the approach and cruise mach when applicable but sometimes they depend on the CG and AoA. For the CG a mid point between the aft and fwd position was taken but for the AoA it was added as an input. 
- Theses coefficients are computed using the convention from Roskam meaning that for the lateral derivatives, the reference length is the wing span. Also, for the derivatives with respect to the rotationnal speed, said speed are made dimensionless by multiplying them by the reference length (MAC or span depending on the speed) and dividing them by two times the airspeed.

This PR also fixes #181 and closes #186

### CODE REVIEW

**Comment by florentLutz** [comment]
Added a commit that resolves #186

File: docs/refs.bib [hunk 1]
```diff
@@ -33,6 +33,13 @@ @book{roskampart5:1985
   publisher={DARcorporation}
 }
 
+@article{roskampart6:1990,
+  title={Airplane Design Part VI, Preliminary Calculation of Aerodynamic Thrust and Power Characteristics. Design},
```
[the remark this answers is not in the document]
Comment (florentLutz) [reply docs/refs.bib:37]: Do not be fooled by appearances, it is complete : https://scholar.google.com/scholar?hl=en&as_sdt=0%2C5&q=roskam+part+vi+1990&btnG=
Well it is referenced this way on Scholar anyway, do we keep it as is ?

File: src/fastga/models/aerodynamics/components/compute_cl_rudder.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+                inputs["data:aerodynamics:rudder:low_speed:Cy_delta_r"] * vt_area / wing_area
+            )
+            outputs["data:aerodynamics:rudder:low_speed:Cl_delta_r"] = (
+                cy_delta_r * (z_v * np.cos(aoa_ref) - lp_vt * np.sin(aoa_ref)) / wing_span
+            )
+        else:
+            # Change of reference surface area
+            cy_delta_r = inputs["data:aerodynamics:rudder:cruise:Cy_delta_r"] * vt_area / wing_area
+            outputs["data:aerodynamics:rudder:cruise:Cl_delta_r"] = (
+                cy_delta_r * (z_v * np.cos(aoa_ref) - lp_vt * np.sin(aoa_ref)) / wing_span
```
[the remark this answers is not in the document]
Comment (florentLutz) [reply src/fastga/models/aerodynamics/components/compute_cl_rudder.py:92]: The reference AoA is an input so it's up to the user to define which case should be considered.

File: src/fastga/models/aerodynamics/components/compute_cl_rudder.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+                "data:aerodynamics:rudder:cruise:Cy_delta_r", val=np.nan, units="rad**-1"
+            )
+            self.add_output("data:aerodynamics:rudder:cruise:Cl_delta_r", units="rad**-1")
+
+        self.declare_partials(of="*", wrt="*", method="exact")
+
+    def compute(self, inputs, outputs, discrete_inputs=None, discrete_outputs=None):
+
+        z_v = (
+            inputs["data:geometry:wing:root:z"]
```
[the remark this answers is not in the document]
Comment (florentLutz) [reply src/fastga/models/aerodynamics/components/compute_cl_rudder.py:69]: This geometric data was added solely for the purpose of computing the aerodynamic derivatives. As such, the convention from Roskam was used. As seen in the description of the variable it represents "Distance from the fuselage center line to the middle of the wing, taken positive when wing is below fuselage center line".

File: src/fastga/models/aerodynamics/components/ht/compute_cm_pitch_rate_ht.py [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+        ht_area = inputs["data:geometry:horizontal_tail:area"]
+        eta_h = inputs["data:aerodynamics:horizontal_tail:efficiency"]
+        volume_coeff_ht = inputs["data:geometry:horizontal_tail:volume_coefficient"]
+        lp_ht = inputs["data:geometry:horizontal_tail:MAC:at25percent:x:from_wingMAC25"]
+
+        # A CG position is necessary for the computation of this coefficient, we will thus assume
+        # a CG between the two extrema
+        x_cg_fwd = inputs["data:weight:aircraft:CG:fwd:x"]
+        x_cg_aft = inputs["data:weight:aircraft:CG:aft:x"]
+        x_cg_mid = (x_cg_fwd + x_cg_aft) / 2.0
```
[the remark this answers is not in the document]
Comment (florentLutz) [reply src/fastga/models/aerodynamics/components/ht/compute_cm_pitch_rate_ht.py:87]: Does that mean that it needs to be changed ? In that case I'd rather have it as an input as is done for the angle of attack.

File: src/fastga/models/geometry/geom_components/wing/components/compute_wing_z.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+import numpy as np
+
+import openmdao.api as om
+import fastoad.api as oad
+
+from ..constants import SUBMODEL_WING_HEIGHT
+
+
+@oad.RegisterSubmodel(SUBMODEL_WING_HEIGHT, "fastga.submodel.geometry.wing.height.legacy")
```
[the remark this answers is not in the document]
Comment (florentLutz) [reply src/fastga/models/geometry/geom_components/wing/components/compute_wing_z.py:23]: See my remark https://github.com/supaero-aircraft-design/FAST-GA/pull/185#discussion_r1016618228

File: src/fastga/models/geometry/unitary_tests/test_cirrus_sr22.py [hunk 6]
```diff
@@ -408,6 +459,20 @@ def test_geometry_wing_y():
...
+def test_geometry_wing_z():
+    """Tests computation of the wing Zs"""
+
+    # Research independent input value in .xml file and add values calculated from other modules
+    ivc = get_indep_var_comp(list_inputs(ComputeWingZ()), __file__, XML_FILE)
+
+    # Run problem and check obtained value(s) is/(are) correct
+    problem = run_system(ComputeWingZ(), ivc)
+    wing_y2 = problem.get_val("data:geometry:wing:root:z", units="m")
+    assert wing_y2 == pytest.approx(0.596, rel=1e-2)
```
[the remark this answers is not in the document]
Comment (florentLutz) [reply src/fastga/models/geometry/unitary_tests/test_cirrus_sr22.py:471]: Same remark as for https://github.com/supaero-aircraft-design/FAST-GA/pull/185#discussion_r1016618228 😛

File: src/fastga/models/aerodynamics/components/compute_cl_aileron.py [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+from .figure_digitization import FigureDigitization
+from ..constants import SUBMODEL_CL_AILERON
+
+
+@oad.RegisterSubmodel(
+    SUBMODEL_CL_AILERON, "fastga.submodel.aerodynamics.aileron.roll_moment.legacy"
+)
+class ComputeClDeltaAileron(FigureDigitization):
+    """
+    Roll moment due to rudder estimated based on the methodology presented in Gudmundsson. This
```
[the remark this answers is not in the document]
Comment (florentLutz) [reply src/fastga/models/aerodynamics/components/compute_cl_aileron.py]: Corrected in 67d6b43

File: src/fastga/models/aerodynamics/components/compute_cl_aileron.py [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+        else:
+            mach = inputs["data:aerodynamics:cruise:mach"]
+
+        aileron_max_deflection = inputs["data:geometry:wing:aileron:max_deflection"]
+        aileron_chord_ratio = inputs["data:geometry:wing:aileron:chord_ratio"]
+
+        aileron_outer_span = wing_span / 2.0
+        aileron_inner_span = wing_span * (1.0 - aileron_span_ratio) / 2.0
+
+        alpha_aileron = self.k_prime_single_slotted(aileron_max_deflection, aileron_chord_ratio)
```
[the remark this answers is not in the document]
Comment (florentLutz) [reply src/fastga/models/aerodynamics/components/compute_cl_aileron.py]: Corrected in 67d6b43

File: src/fastga/models/aerodynamics/components/compute_cy_beta.py [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+
+from ..constants import (
+    SUBMODEL_CY_BETA,
+    SUBMODEL_CY_BETA_WING,
+    SUBMODEL_CY_BETA_VT,
+    SUBMODEL_CY_BETA_FUSELAGE,
+)
+
+
+@oad.RegisterSubmodel(SUBMODEL_CY_BETA, "submodel.aerodynamics.aircraft.side_force_beta.legacy")
```
[the remark this answers is not in the document]
Comment (florentLutz) [reply src/fastga/models/aerodynamics/components/compute_cy_beta.py:27]: Added in 67d6b43 for Cn_beta and Cy_beta

File: src/fastga/models/aerodynamics/unitary_tests/data/beechcraft_76.xml [hunk 10]
```diff
@@ -156,6 +164,11 @@
             </x>
           </leading_edge>
         </MAC>
+        <aileron>
+          <chord_ratio is_input="True">0.26<!--aileron chord as a percentage of the wing chord--></chord_ratio>
+          <max_deflection units="rad" is_input="True">40.0</max_deflection>
```
[the remark this answers is not in the document]
Comment (florentLutz) [reply src/fastga/models/aerodynamics/unitary_tests/data/beechcraft_76.xml]: Corrected in 67d6b43

File: src/fastga/models/geometry/unitary_tests/test_daher_tbm900.py [hunk 11]
```diff
@@ -265,6 +272,20 @@ def test_compute_ht_wet_area():
...
 
+def test_compute_ht_volume_coefficient():
+    """Tests computation of the horizontal tail volume coefficient"""
+
+    # Research independent input value in .xml file and add values calculated from other modules
+    ivc = get_indep_var_comp(list_inputs(ComputeHTVolumeCoefficient()), __file__, XML_FILE)
+
+    # Run problem and check obtained value(s) is/(are) correct
+    problem = run_system(ComputeHTVolumeCoefficient(), ivc)
+    wet_area = problem.get_val("data:geometry:horizontal_tail:volume_coefficient")
```
[the remark this answers is not in the document]
Comment (florentLutz) [reply src/fastga/models/geometry/unitary_tests/test_daher_tbm900.py]: Corrected in 67d6b43

File: src/fastga/models/geometry/unitary_tests/test_cirrus_sr22.py [hunk 12]
```diff
@@ -265,6 +272,20 @@ def test_compute_ht_wet_area():
...
 
+def test_compute_ht_volume_coefficient():
+    """Tests computation of the horizontal tail volume coefficient"""
+
+    # Research independent input value in .xml file and add values calculated from other modules
+    ivc = get_indep_var_comp(list_inputs(ComputeHTVolumeCoefficient()), __file__, XML_FILE)
+
+    # Run problem and check obtained value(s) is/(are) correct
+    problem = run_system(ComputeHTVolumeCoefficient(), ivc)
+    wet_area = problem.get_val("data:geometry:horizontal_tail:volume_coefficient")
```
[the remark this answers is not in the document]
Comment (florentLutz) [reply src/fastga/models/geometry/unitary_tests/test_cirrus_sr22.py]: Corrected in 67d6b43

File: src/fastga/models/geometry/unitary_tests/test_cirrus_sr22.py [hunk 13]
```diff
@@ -265,6 +272,20 @@ def test_compute_ht_wet_area():
...
+    assert wet_area == pytest.approx(0.674, rel=1e-2)
```
[the remark this answers is not in the document]
Comment (florentLutz) [reply src/fastga/models/geometry/unitary_tests/test_cirrus_sr22.py]: Corrected in 67d6b43

File: docs/refs.bib (hunk 1 above)
[the remark this answers is not in the document]
Comment (esnguyenvan) [reply docs/refs.bib:37]: I would use the DARCorporation one, it seems more official (and correctly cited): Roskam, J. (1985). Airplane Design: Part 6-Preliminary Calculation of Aerodynamic, Thrust and Power Characteristics. DARcorporation.

[the remark this answers is not in the document]
Comment (florentLutz) [reply docs/refs.bib]: Done in 2ba9bc0

File: src/fastga/models/aerodynamics/components/compute_cl_rudder.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (esnguyenvan) [reply src/fastga/models/aerodynamics/components/compute_cl_rudder.py:92]: I see. Then, the calculation of the derivative should be done with one set of input for low speed and a different set of input for high_speed (although probably only the AoA changes). This is not a problem but then it argues in favour of a dedicated module, realising the calculation for high or low speed only.

File: src/fastga/models/aerodynamics/components/ht/compute_cm_pitch_rate_ht.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (esnguyenvan) [reply src/fastga/models/aerodynamics/components/ht/compute_cm_pitch_rate_ht.py:87]: Well all other moment coefficients are calculated with respect to the wing_ac (the reference point for all level arms). It is up to the user to transport the moment to the correct CG. As such, I would just set x_cg = x_wing_ac.

File: src/fastga/models/aerodynamics/components/compute_cl_rudder.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (florentLutz) [reply src/fastga/models/aerodynamics/components/compute_cl_rudder.py:92]: Should we take a different value name for the AoA in each case like "settings:aerodynamics:low_speed:reference_flight_conditions:AOA" and "settings:aerodynamics:cruise;reference_flight_conditions:AOA" or keep it like it is now assuming the two conditions won't be ran in the same oad process ?

File: src/fastga/models/geometry/unitary_tests/test_cirrus_sr22.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (esnguyenvan) [reply src/fastga/models/geometry/unitary_tests/test_cirrus_sr22.py:471]: clear

**Comment by florentLutz** [comment]
Left to be done:
- [ ] Change the location of the reference point for lateral derivatives
- [ ] Have two separate variables for the reference AoA for high speed and low speed
- [ ] Isolate derivatives that are not used in the sizing and put them in a separate module to save time

**Review by esnguyenvan — APPROVED** [review]
I approve all changes

**Pull request merged**